### PR TITLE
add sonoff_rf support

### PR DESCRIPTION
**Pull request by mknj**
This PR adds support for the Sonoff RF remote
https://www.itead.cc/wireless-4-buttons-push-cover-remote-controller.html

### CODE REVIEW

**Comment by CurlyMoo** [comment]
Without looking at the commited code, can you start with committed under your own account?

**Comment by mknj** [comment]
@CurlyMoo what do you mean by committed under your own account?

**Comment by CurlyMoo** [comment]
That the commit icon doesn't show the github cat, but your blue icon.

**Comment by mknj** [comment]
fixed

File: libs/pilight/protocols/433.92/sonoff_rf.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			binary[i++] = 1;
+		} 
+	}
+	id = binToDec(binary, 0, 19);
+	id = id & 0xFFFFF;
+	buttons = binToDec(binary, 20, 23);
+	buttons = buttons & 0xF;
+	sonoff_rf->message = json_mkobject();
+	json_append_member(sonoff_rf->message, "id", json_mknumber(id, 0));
+	json_append_member(sonoff_rf->message, "buttonmask", json_mknumber(buttons, 0));
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/sonoff_rf.c]: Why is this not an option defined below.

**Comment by CurlyMoo** [comment]
Will you be adding the sender part as well?

**Comment by mknj** [comment]
currently not. I have no receiver hardware and do not want to release untested code.

File: libs/pilight/protocols/433.92/sonoff_rf.c (hunk 1 above)
Comment (mknj) [reply libs/pilight/protocols/433.92/sonoff_rf.c]: fixed in second commit

**Comment by CurlyMoo** [comment]
Will you be getting receiver hardware anytime soon?

**Comment by orrpan** [comment]
> i guess that i may provide additional code in 4-8 weeks. I ordered a compatible window sensor(sender) and light switch (receiver) in HK.
> i'll add a link to this PR in the forum if someone is interested in this code meanwhile.

I have a "Sonoff DW1 Door&Window Sensor" can send u RAW and test if you implement it.

**Comment by CurlyMoo** [comment]
@mknj any updates?

**Comment by mknj** [comment]
I am using a hardware based solution (sonoff to MQTT) instead of pilight. Please close the pull request

**Comment by orrpan** [comment]
I can try with a receiver

**Pull request closed without merging**